Frame v2 upgrade part2

Neopallium opened the pull request:
Update pallets to Frame v2:

1. Sudo
2. Relayer
3. Protocol Fee
4. Sto
5. Base
6. Bridge
7. Permissions
8. Treasury
9. Group
10. External Agents
11. Corporate Actions
12. Capital Distribution
13. Corporate Ballot
14. Committee
15. PolymeshContracts
16. NFT
17. Balances
18. ComplianceManager
19. Asset
20. Checkpoint
21. Statistics
22. Portfolio
23. Settlement
24. Identity

## changelog

### new features
- Performs pip pruning/refunds on_idle;

### new external API
- Adds the following storage to the pips pallet: `PendingRefunds`,  `VotesToBePruned`; 

### other
- Improves benchmarks, taking into account the number of votes that were pruned/refunded; 
- Adds bounds to the number of pruned votes/refunds;
- Improve Pips pallet documentation
- Support loading chain spec with dev/testnet runtime.
- Add v7.0 chain spec for ci-runtime.
- Add CLI tool to upgrade the ci-chain.
- Run the integration tests before and after a chain upgrade.
- Bump Rust integration resource class.
- Update check_storage_version script for new Frame v2 version.

Neopallium committed:
Fix issue with missing assimilate_storage on Sudo GenesisConfig.

Use workspace deps.

Copy Sudo pallet code from Substrate monthly 2023-03.

cargo fmt Sudo code.

Readd the MIN_WEIGHT support to Sudo.

Update Relayer pallet.

Updated ProtocolFee pallet.

Add missing Rust docs and other cleanup.

Fixup Relayer

Fixup Protocol Fee pallet.

Neopallium commented:
Keeping in WIP until part1 is merged.

HenriqueNogara commented on pallets/relayer/src/lib.rs:
```diff
@@ -72,84 +71,100 @@ pub trait WeightInfo {
+    /// A key can only have one subsidy at a time.  To change subsidisers
+    /// a key needs to call `remove_paying_key` to remove the current subsidy,
+    /// before they can accept a new subsidiser.
+    #[pallet::storage]
```
This storage is missing the getter

Neopallium replied:
Substrate is going to deprecate the getter attribute: https://github.com/paritytech/polkadot-sdk/issues/3326
They have started to remove use of it from the Substrate pallets.  We should start doing the same.  For the v2 upgrade I didn't want to include that change.  This will mostly impact our unit tests and benchmark code.

Neopallium committed:
Frame v2 upgrade part3 (#1773)

Update Sto pallet.

Add 'try-runtime' feature to all pallets.

Update Base pallet.

Update Bridge pallet.

Update Permissions pallet.

Update Treasury pallet.

Upgrade Group pallet.

Use 'Pallet' instead of 'Module' for FRAME v1 pallets too.

Add Frame v2 migration helper.

Support loading chain spec with dev/testnet runtime.

Add v7.0 chain spec for ci-runtime.
Add CLI tool to upgrade the ci-chain.
Run the integration tests before and after a chain upgrade.
Bump Rust integration resource class.

Update check_storage_version script for new Frame v2 version.

Update External Agents pallet.

Update Corporate Actions pallet.

Update Capital Distribution pallet.

Update Corporate Ballot pallet.

Update Committee pallet.

Update PolymeshContracts pallet.

Update NFT pallet.

Updated Balances pallet.

Update ComplianceManager pallet.

Cargo fmt

Update Asset pallet.

Update Checkpoint pallet.

Update Statistics pallet. (#1779)

Update Portfolio pallet.

Update Settlement pallet.

Update Identity pallet.

HenriqueNogara committed:
Pips docs (#1781)

* Improve docs - part 1

* Improve extrinsic documentation

* Add bound to pips (#1784)

* Solve merge conflicts - import path

* Improve insert_live_queue function

* Add unit test; Improve weight calculation (return max between refunds and votes)

* Prune and refund only on_idle

* Remove const and error

Neopallium committed:
Merge branch 'develop' into frame_v2_upgrade_part2
